replay2/memory_remapper/memory_remapper.cc

```diff
@@ -14,4 +14,107 @@
 
 #include "memory_remapper.h"
 
-void foo(int var) { return; }
+namespace agi {
+namespace replay2 {
+
+	void MarkDeadAddressRange(const ReplayAddressRange& replayAddressRange) {
+		const std::byte dead[2] = { std::byte(0xDE), std::byte(0xAD) };
+		for(int i = 0; i < replayAddressRange.length(); ++i) {
+			replayAddressRange.baseAddress().bytePtr()[i] = dead[i %2];
+		}
+	}
+
+	ReplayAddress MemoryRemapper::AddMapping(const MemoryObservation& observation) {
+
+		const CaptureAddress& captureAddress = observation.captureAddress();
+
+		auto replayAddressRangeAndOffset = findReplayAddressRangeAndOffset(captureAddress);
+		if(replayAddressRangeAndOffset.first != nullptr) {
+			throw AddressAlreadyMappedException();
+		}
+
+		const size_t resourceLength = observation.resourceGenerator()->length();
+		if(resourceLength == 0) {
+			throw CannotMapZeroLengthAddressRange();
+		}
+
+		std::byte* replayAllocation = new std::byte [resourceLength];
+		const ReplayAddress replayAddress(replayAllocation);
+
+		observation.resourceGenerator()->generate(replayAddress);
+
+		const CaptureAddressRange captureAddressRange(captureAddress, resourceLength);
+		const ReplayAddressRange replayAddressRange(replayAddress, resourceLength);
+
+		const auto newMapping = std::make_pair(captureAddressRange, replayAddressRange);
```

Comment:
this line is redundant.

I believe:

` captureAddressRanges_.emplace(captureAddressRange, replayAddressRange)`

does the same thing which was the reason that I suggested in the beginning.